catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java

```diff
@@ -312,14 +333,22 @@ private void activateDeactivateExistingConfiguration(String fPidToActivate,
         //Order of disable/enable important. Can't have two enabled configurations with the same
         //id so if there is one to disable, disable it before enabling another one.
         if (configToDisable != null) {
-            fpidToConfigurationMap.remove(configToDisable.getFactoryPid());
+            fpidToConfigurationMap
+                    .remove(configToDisable.getFactoryPid() + configToDisable.getProperties()
+                            .get(BINDING_TYPE));
```

Comment:
Config.getProperties should never return null. The get(BINDING_TYPE) can return null if the bindingType does not exist. In that case it will append the String "null" to the factoryPid as the key.